jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildDockerTask.java

```diff
@@ -88,9 +88,12 @@ public void buildDocker()
       throws InvalidImageReferenceException, IOException, BuildStepsExecutionException,
           CacheDirectoryCreationException, MainClassInferenceException,
           InferredAuthRetrievalException {
-    DockerClient dockerClient =
-        DockerClient.newClient(dockerClientParameters.getExecutable(), null);
-    if (!dockerClient.isDockerInstalled()) {
+    Path dockerExecutable = dockerClientParameters.getExecutable();
+    DockerClient.Builder dockerClientBuilder = new DockerClient.Builder();
+    if (dockerExecutable != null) {
+      dockerClientBuilder.setDockerExecutable(dockerExecutable);
+    }
+    if (!dockerClientBuilder.build().isDockerInstalled()) {
```

Comment:
I just realized - is this creating a `DockerClient` solely for checking the executable exists? I think we can change this to a static method then.

Reply:
Which one do you prefer,
```java
Path executable = dockerClientParameters.getExecutable();
boolean isDockerInstalled =
    executable == null
        ? DockerClient.isDefaultDockerInstalled()
        : DockerClient.isDockerInstalled(executable);
```
or
```java
  DockerClient.isDockerInstalled(dockerClientParameters.getExecutable());
```
?
I guess you don't like passing `null` to mean using a default value, so the former which needs adding an additional static method?

Reply:
I think since ideally `getExecutable` should return `Optional` and perform the logic based on `Optional.isPresent`. Passing `null` down in general feels like tailoring an invoked method's semantics to the semantics of the input (which we happen to do in `BuildConfiguration.Builder` - for example, by allowing `null` to mean default, we lose the ability to actually set something to `null`). I think in this case, both are fine mostly because this is a static method and doesn't pass the nullable value down further (to instantiate instances).

Reply:
Will opt for the first (with two static methods).

> by allowing `null` to mean default, we lose the ability to actually set something to `null`

On a side note, regarding this topic, here's my 2 cents: we shouldn't confuse the semantics of "something missing" (often coded by the `null` value of the Java language or `Optional.isEmpty()` in implementation detail) vs "setting a special, concrete value, e.g., setting a container metadata property to the JSON null value." A component can be designed to act reasonably or properly when something is missing, and the involved parties should agree upon the contract of what will happen in that case. (If "something missing" should translate to setting a property in JSON to the JSON null value, it can work like that by that design.) And if someone does intent to set a property to the JSON null value, then I think it is actually directing to set an explicit value, which is different from saying "I have no information". The confusion may just be that this JSON null value happens to look similar to the `null` keyword in Java code, which may be used to represent that something is missing as one possible Java implementation detail.

Reply:
Much agreed